### PR TITLE
[WIP] AOT compilation 

### PR DESCRIPTION
**Pull request by sendilkumarn**
For AOT to work, we need to remove vendor configuration of webpack 

I was able to achieve 3xxkb (still on higher side 😛)

Still to do.
- [ ] Move out common functionalities (or) revamp webpack common to remove vendor related entries.
- [ ] Images are not currently loading (but the above bandwidth is measured including images)

### CODE REVIEW

File: generators/client/templates/angular/_tsconfig-aot.json [hunk 1]
```diff
@@ -17,29 +17,17 @@
...
-    },
-
-    "files": [
-        "<%= MAIN_SRC_DIR %>app/app.module.ts",
-        "<%= MAIN_SRC_DIR %>app/app.main.ts"
-    ],
-
-    "angularCompilerOptions": {
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
```
Comment (deepu105) [inline generators/client/templates/angular/_tsconfig-aot.json]: 4 space

File: generators/client/templates/angular/webpack/_webpack.prod.js [hunk 2]
```diff
@@ -16,26 +16,143 @@
  See the License for the specific language governing permissions and
  limitations under the License.
 -%>
-const commonConfig = require('./webpack.common.js');
```
Comment (deepu105) [inline generators/client/templates/angular/webpack/_webpack.prod.js]: why is this merged into this file? isnt this common between prod and dev

File: generators/client/templates/angular/webpack/_webpack.prod.js [hunk 3]
```diff
@@ -16,26 +16,143 @@
...
+        ]
+    },
+    entry: {
+        'polyfills': './src/main/webapp/app/polyfills',
+        'global': './src/main/webapp/content/css/global.css',
+        'main': './src/main/webapp/app/app.main'
+    },
     output: {
-        path: path.resolve('./<%= BUILD_DIR %>www'),
+        path: path.resolve('./build/www'),
```
Comment (deepu105) [inline generators/client/templates/angular/webpack/_webpack.prod.js]: template?

**Comment by sendilkumarn** [comment]
I forgot that we have cli here. Sorry closing this

**Pull request closed without merging**